src/index.js

```diff
@@ -55,16 +67,19 @@ export default function supernova(env) {
         if (layout && model) {
           return treeTransform({ layout, model }).then(transformed => {
             setDataTree(transformed);
+            setStyling(stylingUtils.cardStyling({ Theme, layout }));
           });
         }
         return Promise.resolve();
       }, [layout, model]);
 
-      // Should avoid dependency on layout -> store that on dataTree instead?
-      // Should move element dependency to preRender function
-      useEffect(callRender, [layout, element, dataTree]);
-      // Should be bundled with dataTree dependency once layout and element has been moved?
-      useEffect(callRender, [rect]);
+      // This one can split up. Only need to get new height/width when rect is changed
+      useEffect(callPreRender, [element, dataTree, rect.width, rect.height]);
```

Comment:
is putting just rect here not enough?

Reply:
hmm it got caught in an endless loop before, but works using only rect now